### PR TITLE
DPE-5627 Arch support and CC006 compliance

### PR DESCRIPTION
**Pull request by paulomach**
Multi arch support TF "bundle"
Sources charm terraform from charm repo.

Future TODO:
* ~~Parametrized certificates provider (now bundling self-signed)~~
* Renovate rules for update charm revisions

### CODE REVIEW

File: terraform/revisions.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+locals {
+  mysql_revisions = {
+    amd64 = 180, # renovate tag
```
Comment (paulomach) [inline terraform/revisions.tf:3]: Renovate rules would apply here

File: terraform/applications.tf [hunk 2]
```diff
@@ -0,0 +1,87 @@
+module "mysql" {
+  source          = "git::https://github.com/canonical/mysql-k8s-operator//terraform?ref=feature/tf-module"
```
Comment (paulomach) [inline terraform/applications.tf:2]: once merged:

```suggestion
  source          = "git::https://github.com/canonical/mysql-k8s-operator//terraform?ref=main"
```

File: terraform/revisions.tf (hunk 1 above)
Comment (taurus-forever) [reply terraform/revisions.tf:3]: CC: @lucasgameiroborges

**Review by taurus-forever — APPROVED** [review]

File: terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+| `enable_tls` | Whether to enable TLS for the MySQL operator | `bool` | `true` | no |
+| `model_name` | Name of the model to create/use | `string` | `null` | yes |
+| `mysql_backup_access_key` | Backup access key for the MySQL operator | `string` | `""` | yes |
+| `mysql_backup_bucket_name` | Backup bucket for the MySQL operator | `string` | `""` | yes |
+| `mysql_backup_endpoint` | Backup endpoint for the MySQL operator | `string` | `""` | yes |
+| `mysql_backup_region` | Backup region for the MySQL operator | `string` | `""` | yes |
+| `mysql_backup_secret_key` | Backup secret key for the MySQL operator | `string` | `""` | yes |
+| `mysql_charm_channel` | Charm channel for the MySQL operator | `string` | `"8.0/stable"` | no |
+| `mysql_charm_config` | Configuration options for the MySQL operator | `map(string)` | `{}` | no |
+| `mysql_charm_revision` | Charm revision number override for MySQL | `number` | `null` | no |
```
[the remark this answers is not in the document]
Comment (paulomach) [reply terraform/README.md:30]: Yes, unless overridden

File: terraform/revisions.tf (hunk 1 above)
Comment (shayancanonical) [reply terraform/revisions.tf:3]: would all the revision pins need a `# renovate tag` comment?

Comment (paulomach) [reply terraform/revisions.tf:3]: that's a placeholder because I don´t know how to do it, forgot to remove

**Pull request merged**